Tests/MoyaTests/TaskSpec.swift

```diff
@@ -0,0 +1,91 @@
+import Quick
+@testable import Moya
+import Nimble
+import Foundation
+import Alamofire
+
+final class TaskSpec: QuickSpec {
+
+    override func spec() {
+
+        let encodable: [String: String] = ["Hello": "Moya"]
```

Comment:
can we update the test to cover for `[String: Any]`? e.g. `["a": "b", "c": 1]`

Reply:
For the dictionary to be `Encodable` the value must also be `Encodable`. `[String: Any]` won't be encodable.
Having a test with both String and Int we would have to use `[String: AnyEncodable]`, but I'm not sure that what you want.

Reply:
hmm, so currently if we want to pass `[String: Any]` as parameters we would need to convert each value to `AnyEncodable`? what I'm thinking is that we currently allow the parameters to be `[String: Any]` and ideally we would like to keep it this way

Reply:
@amaurydavid did you have a chance to think about the possible `[String: Any]` workaround?

Reply:
Sorry I didn't see your reply.
If think you already highlighted the situation [in this comment](https://github.com/Moya/Moya/pull/1936#pullrequestreview-314164520).

Using `Endocable` is actually enforced by the new parameter encoding from Alamofire, it's not a decision on my part. There have been some questions in the Alamofire github [here](https://github.com/Alamofire/Alamofire/issues/2763#issuecomment-476837964), [there](https://github.com/Alamofire/Alamofire/issues/2991#issuecomment-549216841), and surely in some other places.

It doesn't seem to be any way to use the dict `["a": "b", "c": 1]` into an `Encodable`. There are 2 workarounds for this:

- to replace the dict with a specific type:

```
struct MyParameters: Encodable {
 let a = "b"
 let c = 1
}
```

 - to wrap all values into `AnyEncodable`

Reply:
yeah, so from my personal experience I rarely use `Encodable`/separate structs for a simple API call and there are many places in which I want to pass [String: Any] as parameters

do you think we could create a helper function, e.g. `toEncodable()`, that could wrap `[String: Any]` to `Encodable`? ideally we wouldn't have to do such things, but as I see it right now it's hard with current design

cc @Moya/contributors

Reply:
Instead of a helper func we could had new inits to handle [String: Any] and ignore all values not being Encodable, but it would introduce a behavior users won't expect where some parameters won't be the request while passed to Moya.
I would rather let users make their own `Encodable`.